samples: buck: rework BUCK example into 4 separate ones (#74)
NordicSemiconductor/npmx-zephyr: samples/buck_output_advanced/README.rst

.. _buck_output_advanced_sample:

BUCK - set advanced output voltage
##################################

.. contents::
   :local:
   :depth: 2

> The BUCK Output Advanced sample demonstrates how to control PMIC BUCKs using npmx drivers.

Reviewer: Please use the same sentence structure for all 4 samples.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

The sample also requires an nPM1300 EK.

Overview
********

This sample uses the npmx BUCK driver to set advanced output voltage for the buck converter.

> The BUCK1 starts from voltage 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`).
> The output voltage increases from the selected voltage to the maximum one (3.3 V).

Reviewer: ```suggestion
The BUCK1 starts from 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`), and the output voltage increases gradually every 200 ms to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_MAX`).
```

> This sample uses the npmx BUCK driver to set advanced output voltage for the buck converter.
> The BUCK1 starts from voltage 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`).
> The output voltage increases from the selected voltage to the maximum one (3.3 V).
> 
> Voltage can be changed only for BUCK1.
> BUCK2 voltage is set by default to 3.0 V.

Reviewer: ```suggestion
This sample uses the npmx BUCK driver to set advanced output voltage for the buck converter.'

The output voltage for BUCK1 starts from 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`) and increases by the selected value to the maximum 3.3 V.
Voltage can be changed only for BUCK1.
BUCK2 voltage is set by default to 3.0 V.
```
Please check if correct.

Wiring
******

> #. Connect the TWI interface between the chosen DK and the nPM1300 EK as in the following table:

Reviewer: ```suggestion
1. Connect the TWI interface between the chosen DK and the nPM1300 EK as in the following table:
```
Both versions are technically correct but we decided to use this variant for numbering.

   .. list-table:: nPM1300 EK connections.
     :widths: 25 25 25 25
     :header-rows: 1

     * - nPM1300 EK pins
       - nRF5340 DK pins
       - nRF52840 DK pins
       - nRF52832 DK pins
     * - GPIO0
       - P1.10
       - P1.10
       - P0.24
     * - SDA
       - P1.02
       - P0.26
       - P0.26
     * - SCL
       - P1.03
       - P0.27
       - P0.27
     * - VOUT2 & GND
       - External supply (P21)
       - External supply (P21)
       - External supply (P21)

#. Make the following connections on the nPM1300 EK:

   * Connect a USB power supply to the **J3** connector.
   * On the **P18** pin header, connect **VOUT2** and **VDDIO** pins with a jumper.
   * On the **P17** pin header, connect all LEDs with jumpers.
   * On the **P13** pin header, connect **VSET1** and **GND** pins with a jumper.
   * On the **P14** pin header, connect **VSET2** and **RSET2** pins with a jumper.

#. Make the following connections on the chosen DK:

   * Set the **SW9** nRF power source switch to **VDD**.
   * Set the **SW10** VEXT → nRF switch to **ON**.

Configuration
*************

|config|

Building and running
********************

> .. |sample path| replace:: :file:`samples/buck`

Reviewer 2: ```suggestion
.. |sample path| replace:: :file:`samples/buck_output_advanced`
```

.. include:: /includes/build_and_run.txt

Testing
=======

|test_sample|

#. Connect a multimeter to VOUT1 that is the output voltage pin of BUCK1.
#. |connect_kit|
#. |connect_terminal|

> #. Check output voltage on EK **VOUT1** pin - it should start from 1.0 V and increase by 200 ms up to 3.3 V.

Reviewer: ```suggestion
#. Check output voltage on EK **VOUT1** pin - it should start from 1.0 V and increase every 200 ms up to 3.3 V.
```

#. Verify output voltage on BUCK1.

Dependencies
************

> This sample uses drivers from npmx.

Reviewer: ```suggestion
This sample uses drivers from npmx.
```
Would be great if we can be more specific (discussed offline).

.. _buck_output_advanced_sample:

BUCK - Advanced output voltage setting
######################################

.. contents::
   :local:
   :depth: 2

This BUCK sample variant demonstrates how to set advanced output voltage with the use of npmx drivers.

Requirements
************

The sample supports the following development kits:

.. table-from-sample-yaml::

The sample also requires an nPM1300 EK.

Overview
********

This sample uses the npmx BUCK driver to set advanced output voltage for the buck converter.

> The BUCK1 starts from 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`), and the output voltage increases gradually every 200 ms to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_MAX`).
> Voltage can be changed only for BUCK1.

Reviewer: ```suggestion
The BUCK1 starts from 1.0 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_1V0`), and the output voltage increases gradually every 200 ms to 3.3 V (:c:enumerator:`NPMX_BUCK_VOLTAGE_MAX`).

Voltage can be changed only for BUCK1.
```
